### PR TITLE
Add E0500 error explanation

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @Manishearth 

Part of #32777.

### CODE REVIEW

File: src/librustc_borrowck/diagnostics.rs [hunk 1]
```diff
@@ -378,6 +378,53 @@ let c = &i; // still ok!
...
+
+Or, if the type implements the `Clone` traits, you can clone it between
+closures:
+
+```
+fn you_know_nothing(john_snow: &mut i32) {
+    let mut john_copy = john_snow.clone();
+    let nights_watch = || {
+        john_copy = 2;
+    };
```
Comment (manfredbrandl) [inline src/librustc_borrowck/diagnostics.rs]: *john_copy = 2;
missing *

Comment (GuillaumeGomez) [inline src/librustc_borrowck/diagnostics.rs]: Indeed, thanks!

Comment (GuillaumeGomez) [inline src/librustc_borrowck/diagnostics.rs]: I answered too quickly: there is no need for the Deref use here.

File: src/librustc_borrowck/diagnostics.rs [hunk 2]
```diff
@@ -378,6 +378,53 @@ let c = &i; // still ok!
 ```
 "##,
 
+E0500: r##"
+A borrowed variable was used in another closure. Example of erroneous code:
+
+```compile_fail
+fn you_know_nothing(john_snow: &mut i32) {
```
Comment (birkenfeld) [inline src/librustc_borrowck/diagnostics.rs]: IIRC, s/john/jon/ :)

Comment (GuillaumeGomez) [inline src/librustc_borrowck/diagnostics.rs]: Unforgivable! Thanks a lot!

**Comment by GuillaumeGomez** [comment]
cc @steveklabnik

File: src/librustc_borrowck/diagnostics.rs [hunk 3]
```diff
@@ -378,6 +378,53 @@ let c = &i; // still ok!
...
+            *jon_snow = 2;
+        };
+    } // At this point, `jon_snow` is free.
+    let starks = || {
+        *jon_snow = 3;
+    };
+}
+```
+
+Or, if the type implements the `Clone` traits, you can clone it between
```
Comment (steveklabnik) [inline src/librustc_borrowck/diagnostics.rs]: trait, not traits

**Pull request merged**